### PR TITLE
Add pretrained weight for autoformer

### PR DESCRIPTION
**Pull request by crj1998**
### Description ###
The AutoFormer in `hub` is
Implemented the pre-trained weights load function(`load_searched_model`)  of Autoformer in `hub`. This `load_searched_model` method will load the searched model converted from [ official  Autoformer repo](https://github.com/microsoft/Cream/tree/83a154beb6f85dd6141853b4b7c0738eeec628ba/AutoFormer). The results were consistent.
reproduce

#### key modification

- add pretrained url in  
https://github.com/microsoft/nni/blob/d5921a7312da8c12199e1acb0faf090bebce2b7c/nni/retiarii/hub/pytorch/utils/pretrained.py#L14
- use the fixed head dim rather than 
https://github.com/microsoft/nni/blob/d5921a7312da8c12199e1acb0faf090bebce2b7c/nni/retiarii/hub/pytorch/autoformer.py#L94-L106  

#### Test Options ####
  - [x] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [x] test case
  - [ ] doc

### How to test ###

```
import nni.retiarii.hub.pytorch as hub

# support name: 'autoformer-tiny', 'autoformer-small', 'autoformer-base' 
name = "autoformer-tiny"
model = hub.AutoformerSpace.load_searched_model(name, pretrained=True, download=True)
model.eval()

evaluate(model, dataloader)
```

### CODE REVIEW

File: nni/retiarii/hub/pytorch/utils/pretrained.py [hunk 1]
```diff
@@ -37,6 +37,11 @@
 
     # spos
     'spos': f'{NNI_BLOB}/nashub/spos-0b17f6fc.pth',
+
+    # Autoformer
+    "autoformer-tiny": f'{NNI_BLOB}/nashub/autoformer_tiny-7bd6a6f4.pth',
```
Comment (ultmaster) [inline nni/retiarii/hub/pytorch/utils/pretrained.py:42]: Suggest `autoformer-tiny` rather than `autoformer_tiny`.

File: nni/retiarii/hub/pytorch/utils/pretrained.py [hunk 2]
```diff
@@ -37,6 +37,11 @@
 
     # spos
     'spos': f'{NNI_BLOB}/nashub/spos-0b17f6fc.pth',
+
+    # Autoformer
```
Comment (ultmaster) [inline nni/retiarii/hub/pytorch/utils/pretrained.py:41]: Please add the information about checkpoints in `hub/README.md`.

File: nni/retiarii/hub/pytorch/autoformer.py [hunk 3]
```diff
@@ -92,27 +90,27 @@ def __init__(
             rpe_length=14) -> None:
         super().__init__()
         self.num_heads = num_heads
-        head_dim = embed_dim // num_heads
+        # head_dim = embed_dim // num_heads
+        head_dim = 64
```
Comment (ultmaster) [inline nni/retiarii/hub/pytorch/autoformer.py:94]: Why fixed?

File: nni/retiarii/hub/pytorch/autoformer.py [hunk 4]
```diff
@@ -124,7 +122,8 @@ def forward(self, x):
         attn = attn.softmax(dim=-1)
         attn = self.attn_drop(attn)
 
-        x = (attn @ v).transpose(1, 2).reshape(B, N, C)
+        x = (attn @ v).transpose(1, 2).reshape(B, N, -1)
+
```
Comment (ultmaster) [inline nni/retiarii/hub/pytorch/autoformer.py:126]: Please fix lint.
